### PR TITLE
[ML] Add loss_function to regression

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Adds parameters `loss_function` and `loss_function_parameter`
to regression.

### CODE REVIEW

**Review by benwtrent — COMMENTED** [review]
I have an overall design question.

To my knowledge, only the `huber` function will actually use the lossFunctionParameter. It almost seems to me that this could be a named object. 

If we don't want the complexity a named object requires, what do you think of adding validations? We shouldn't allow the `lossFunctionParameter` unless it is being used.

**Comment by dimitris-athanasiou** [comment]
`msle` also uses the parameter. I also had the same concern and I discussed it with @valeriy42. Valeriy is confident we won't be introducing new loss functions in which case it probably makes sense to keep this simple and simply ignore the parameter for `mse`.

**Review by benwtrent — APPROVED** [review]

**Comment by valeriy42** [comment]
@benwtrent you are right, in a canonical form MSLE does not require a parameter, however, @tveasey had a useful insight that `log(a+1)` in the MSLE formulation actually defines the "area of interest" for the loss function. We are going to change it to `log(a+t)` to be able to "focus" the loss function where we need it.

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 1]
```diff
@@ -262,6 +262,15 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=gamma]
 (Optional, double) 
 include::{docdir}/ml/ml-shared.asciidoc[tag=lambda]
 
+`loss_function`::::
+(Optional, string)
+The loss function used during regression. Available options are `mse` (Mean Squared Error),
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:267]: I don't think this requires capitalization
```suggestion
The loss function used during regression. Available options are `mse` (mean squared error),
```

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 2]
```diff
@@ -262,6 +262,15 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=gamma]
...
+`msle` (Mean Squared Logarithmic Error),  `huber` (Pseudo-Huber loss). Defaults to `mse`.
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:268]: Ditto re mean squared logarithmic, whereas huber seems to be a person's name and can be capitalized
```suggestion
`msle` (mean squared logarithmic error),  `huber` (Pseudo-Huber loss). Defaults to `mse`.
```

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 3]
```diff
@@ -262,6 +262,15 @@ include::{docdir}/ml/ml-shared.asciidoc[tag=gamma]
...
+
+`loss_function_parameter`::::
+(Optional, double)
+A strictly positive number that is used as a parameter to the `loss_function`.
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:272]: > A strictly positive number ...

Why is "strictly" required here?

Comment (dimitris-athanasiou) [reply docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:272]: Fair point! I'll remove it.

**Comment by dimitris-athanasiou** [comment]
@elasticmachine update branch

**Pull request merged**